test/index.test.js

```diff
@@ -95,86 +95,94 @@ const mocks = {
 
 describe('GoodInflux', () => {
     describe('HTTP URL =>', () => {
-        it('Sends events in a stream to HTTP server', (done) => {
-            const server = mocks.getHttpServer(expectedMsgResponse, done);
-            const stream = mocks.readStream();
-
-            server.listen(0, '127.0.0.1', () => {
-                const reporter = new GoodInflux(mocks.getUri(server, 'http'), {
-                    threshold: 5
+        it('Sends events in a stream to HTTP server', () => {
+            return new Promise((resolve) => {
```

Comment:
If you have not done so already, might be worth breaking this on purpose to make sure it fails. Kind of spaghetti code (and probably my fault 😸 )

A good future refactor would be to pass the `reject` param in as well and call that on assertion failure. As is I suspect this will time out the Promise.